net/bridge.go

```diff
@@ -108,3 +199,307 @@ func FindBridgeIP(bridgeName string, subnet *net.IPNet) (net.IP, error) {
 	// No subnet, or none in the required subnet; just return the first one
 	return netdev.CIDRs[0].IP, nil
 }
+
+type BridgeConfig struct {
+	DockerBridgeName string
+	WeaveBridgeName  string
+	DatapathName     string
+	NoFastdp         bool
+	NoBridgedFastdp  bool
+	AWSVPC           bool
+	NPC              bool
+	MTU              int
+	Mac              string
+	Port             int
+}
+
+func CreateBridge(procPath string, config *BridgeConfig) (Bridge, error) {
+	bridgeType, err := DetectBridgeType(config.WeaveBridgeName, config.DatapathName)
+	if err != nil {
+		return nil, err
+	}
+
+	if bridgeType == nil {
+		bridgeType = bridgeImpl{}
```

Comment:
This function is relatively hard to reason about:
- we do more than "one thing", e.g. create bridges, modify configuration, create datapath, handle special cases related to AWSVPC, configure `iptables` rules, etc.
- the succession of `bridgeType` overrides matches `try_create_bridge`'s logic, but is relatively hard to follow

Suggestion: 
- have simple conditions to choose the bridge implementation
- give more responsibilities to the bridge implementation
- split this function in smaller parts, each responsible for one concern, or delegate to the bridge implementation (e.g. would it make sense to have the creation of the datapath happening as part of the instantiation of the bridge impl.?)

Essentially, L222-244 do this (approximate refactoring, double-negation removed), which could be easier to follow:
```
func (...) defaultBridgeType(...) (...) {
    if config.Fastdp && odp.IsSupported() {
        if config.BridgedFastdp {
            return bridgedFastdpImpl{datapathName: config.DatapathName}, nil
        } else {
            return fastdpImpl{datapathName: config.WeaveBridgeName}, nil
        }
    } else {
        return bridgeImpl{}, nil
    }
}
[...]
if bridgeType == nil {
    bridgeType = (...).defaultBridgeType(...)
    if err := bridgeType.init(config); err != nil {
        return nil, err
    }
    if err := configureIPTables(config); err != nil {
        return bridgeType, errors.Wrap(err, "configuring iptables")
    }
}
[...]
```